lib/puppet/application/resource.rb

```diff
@@ -209,6 +216,15 @@ def parse_args(args)
     [type, name, params]
   end
 
+  def create_default_environment!
```

Comment:
Not blocking, but the `!` makes me think this method creates an environment on disk

Reply:
Hm okay, I added it because we modify the value of `Puppet[:environment]` so it has some side effects, but I agree it doesn't really warrant a bang so I'll remove it.